pkg/okteto/secrets.go

```diff
@@ -84,7 +85,7 @@ func (c *userClient) GetContext(ctx context.Context) (*types.UserContext, error)
 		} `graphql:"credentials(space: $cred)"`
 	}
 	variables := map[string]interface{}{
-		"cred": graphql.String(""),
+		"cred": graphql.String(os.Getenv("OKTETO_NAMESPACE")),
```

Comment:
@jLopezbarb @maroshii this will work for pipelines. For Okteto CLI running in the laptop, `OKTETO_NAMESPACE` won't be available and we need to infer the right manifest, or the control plane won't be able to redirect to the right worker cluster

Reply:
@jLopezbarb is it safe to get the namespace from the okteto context? or it's not initialized yet?

Reply:
@pchico83 yeah this is a stopgap solution, we discuss it this morning. This is simply to unblock the deploy from the ui and `okteto pipeline deploy`. With this change we don't recreate the default namespace in the worker. A better fix is still required

Reply:
@pchico83 the real issue that we need to fix, but we will probably need to think about it is that the function `AddOktetoCredentialsToCfg` it is not using the namespace retrieved in the credentials https://github.com/okteto/okteto/blob/0865499c19409a7c41b9b94b03d31ce7267fbde6/pkg/okteto/context.go#L360.

So even if we set here the right namespace, we are not using it to create the Kubeconfig. This works in a single multicluster approach because all the namespaces are in the same cluster, but it won't work with multicluster.

As Fran said, this is a quick fix for the installer issue, as the credentials query will be done with the right namespace and it shouldn't affect other executions, as the namespaces should be in the same cluster for current users and the function `AddOktetoCredentialsToCfg` is still using the same value

Reply:
@maroshii okteto_namespace is not fixed there, but the proper fix would be to add a namespace variable to the function `getUserContext` which is called from the `getLoggedUserContext` function that contains `ctxOptions`. `ctxOptions` have the namespace if it's a command that will need it such as okteto deploy. Probably is that what you want

Reply:
@jLopezbarb you mean we should do that as part of this PR or is that part of the bigger/better fix we should do that Nacho is describing?

Reply:
I'd do it as part of this PR since that would cover most of the cases, I can't ensure that `OKTETO_NAMESPACE` env var is set

Reply:
The installer always sets it so we can know for sure that it's there in those jobs which is what we are looking for to being with. It actually feels more cumbersome to me to cover _most_ of the cases and not all if we still need a follow up PR but I'm not too familiar with the cli codebase. Happy to make the changes you are suggesting but we'll need to sync since I'm not sure how to approach a more complete change.

Reply:
@maroshii I think Javi means to do this

```
diff --git a/cmd/context/create.go b/cmd/context/create.go
index 6e268375..1e8d63e7 100644
--- a/cmd/context/create.go
+++ b/cmd/context/create.go
@@ -1,4 +1,4 @@
-// Copyright 2023 The Okteto Authors
+// Copyright 2022 The Okteto Authors
 // Licensed under the Apache License, Version 2.0 (the "License");
 // you may not use this file except in compliance with the License.
 // You may obtain a copy of the License at
@@ -273,7 +273,7 @@ func getLoggedUserContext(ctx context.Context, c *ContextCommand, ctxOptions *Co

        okteto.Context().Token = user.Token

-       userContext, err := c.getUserContext(ctx)
+       userContext, err := c.getUserContext(ctx, ctxOptions.Namespace)
        if err != nil {
                return nil, err
        }
@@ -310,7 +310,7 @@ func (*ContextCommand) initKubernetesContext(ctxOptions *ContextOptions) error {
        return nil
 }

-func (c ContextCommand) getUserContext(ctx context.Context) (*types.UserContext, error) {
+func (c ContextCommand) getUserContext(ctx context.Context, ns string) (*types.UserContext, error) {
        client, err := c.OktetoClientProvider.Provide()
        if err != nil {
                return nil, err
@@ -318,7 +318,7 @@ func (c ContextCommand) getUserContext(ctx context.Context) (*types.UserContext,

        retries := 0
        for retries <= 3 {
-               userContext, err := client.User().GetContext(ctx)
+               userContext, err := client.User().GetContext(ctx, ns)

                if err != nil && oktetoErrors.IsForbidden(err) {
                        if err := c.OktetoContextWriter.Write(); err != nil {
diff --git a/pkg/okteto/secrets.go b/pkg/okteto/secrets.go
index 25703f4c..878bb316 100644
--- a/pkg/okteto/secrets.go
+++ b/pkg/okteto/secrets.go
@@ -56,7 +56,7 @@ func (c *OktetoClient) GetSecrets(ctx context.Context) ([]types.Secret, error) {
 }

 // GetSecrets returns the secrets from Okteto API
-func (c *userClient) GetContext(ctx context.Context) (*types.UserContext, error) {
+func (c *userClient) GetContext(ctx context.Context, ns string) (*types.UserContext, error) {
        var queryStruct struct {
                User struct {
                        Id              graphql.String
@@ -84,7 +84,7 @@ func (c *userClient) GetContext(ctx context.Context) (*types.UserContext, error)
                } `graphql:"credentials(space: $cred)"`
        }
        variables := map[string]interface{}{
-               "cred": graphql.String(""),
+               "cred": graphql.String(ns),
        }
        err := query(ctx, &queryStruct, variables, c.client)
        if err != nil {
diff --git a/pkg/types/interface.go b/pkg/types/interface.go
index 0f3d989e..84dbaea6 100644
--- a/pkg/types/interface.go
+++ b/pkg/types/interface.go
@@ -29,7 +29,7 @@ type OktetoInterface interface {

 // UserInterface represents the client that connects to the user functions
 type UserInterface interface {
-       GetContext(ctx context.Context) (*UserContext, error)
+       GetContext(ctx context.Context, ns string) (*UserContext, error)
 }

 // NamespaceInterface represents the client that connects to the namespace functions
```

If the environment variable `OKTETO_NAMESPACE` is set, `ctxOptions.Namespace` will contain that. Can you confirm @jLopezbarb ?

Reply:
Great, got it. Thanks @ifbyol!. Updated